src/Features/CSharp/Portable/InlineParameterNameHints/InlineParamNameHintsService.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.InlineParameterNameHints;
+using Microsoft.CodeAnalysis.Text;
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.CSharp.InlineParameterNameHints
+{
+    /// <summary>
+    /// The service to locate the positions in which the adornments should appear
+    /// as well as associate the adornments back to the parameter name
+    /// </summary>
+    [ExportLanguageService(typeof(IInlineParamNameHintsService), LanguageNames.CSharp), Shared]
+    internal class InlineParamNameHintsService : IInlineParamNameHintsService
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public InlineParamNameHintsService()
+        {
+        }
+
+        public async Task<IEnumerable<NameAndSpan>> GetInlineParameterNameHintsAsync(
+            Document document,
+            TextSpan textSpan,
+            CancellationToken cancellationToken)
+        {
+            var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var node = await tree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+            var spans = new List<NameAndSpan>();
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var invocations = node.DescendantNodes().OfType<InvocationExpressionSyntax>();
+
+            NameAndSpan nameAndSpan;
+            foreach (var invo in invocations)
+            {
+                foreach (var argument in invo.ArgumentList.Arguments)
+                {
+                    if (argument.NameColon == null && IsLiteralOrNoNamedExpression(argument))
```

Comment:
IsLiteralOrNoNamedExpression feels redundant.  literals woudl always be 'NoNamedExpression's right?   Maybe just `IsExpressionThatDoesNotHaveName` or IsExpressionThatDoesNotEndWIthName`? (not sure your precise logic, but those read a lot better to me).